scripts/cluster_buildings.py

```diff
@@ -2,152 +2,144 @@
+        if row.geometry.type == "Polygon":
+            features.append(
+                {
+                    "properties": {
+                        "id": row.id,
+                        "tags_building": row.tags_building,
+                    },
+                    "geometry": row.geometry,
+                }
+            )
+    buildings_geodataframe = gpd.GeoDataFrame.from_features(features)
+    microgrid_buildings = microgrid_buildings.to_crs(crs)
+    area_m2 = microgrid_buildings.geometry.area
+    area_m2 = pd.Series(area_m2)
+    buildings_geodataframe["area_m2"] = area_m2
+    for index, row in buildings_geodataframe.iterrows():
+        if row.tags_building == "yes":
+            if row.area_m2 < house_area_limit:
+                buildings_geodataframe.loc[index, "tags_building"] = "house"
+            else:
+                buildings_geodataframe.loc[index, "tags_building"] = "yes"
+    return buildings_geodataframe
+    # buildings_geodataframe.to_file(output_file)
+
+
+def get_central_points_geojson(
+    input_filepath, output_filepath, n_clusters, crs, house_area_limit
+):
+    microgrid_buildings = buildings_classification(
+        input_filepath, crs, house_area_limit
+    )
+    # microgrid_buildings = gpd.read_file(input_filepath)
+    centroids_building = [
+        (row.geometry.centroid.x, row.geometry.centroid.y)
+        for row in microgrid_buildings.itertuples()
+    ]
+    microgrid_buildings["centroid_coordinates"] = centroids_building
+
+    centroids_building = np.array(centroids_building)
+    kmeans = KMeans(n_clusters=n_clusters, random_state=0).fit(centroids_building)
     centroids = kmeans.cluster_centers_
 
-    # Select most central point in each cluster
     central_points = []
     for i in range(kmeans.n_clusters):
-        cluster_points = points[kmeans.labels_ == i]
+        cluster_points = centroids_building[kmeans.labels_ == i]
         distances = np.linalg.norm(cluster_points - centroids[i], axis=1)
         central_point_idx = np.argmin(distances)
         central_points.append(cluster_points[central_point_idx])
 
-    # Create GeoJSON feature for each central point
-    features = []
+    central_features = []
     for i, central_point in enumerate(central_points):
-        feature = {
-            "type": "Feature",
-            "geometry": {"type": "Point", "coordinates": central_point.tolist()},
-            "properties": {"cluster": i},
-        }
-        features.append(feature)
-
-    # Create GeoJSON object with all central point features
-    geojson = {"type": "FeatureCollection", "features": features}
-
-    # Write central point GeoJSON data to file
-    with open(output_filepath, "w") as f:
-        json.dump(geojson, f)
+        central_features.append(
+            {
+                "geometry": Point(central_point),
+                "cluster": i,
+            }
+        )
+    central_features = gpd.GeoDataFrame(central_features)
```

Comment:
specify the crs of the other dataframe to avoid crs issues; you can do:
`gpd.GeoDataFrame(central_features, crs=microgrid_buildings.crs)`

Reply:
Perfect!